### PR TITLE
Refactor player #33

### LINKED ISSUE #33: Create phaser quest inspired player
Refactor Player.js class

### PR DESCRIPTION
**Pull request by abc516**
Closes #33

### CODE REVIEW

File: app/states/playerRefactor.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    // Send context as first argument!!
+    this.anchor.set(0.25, 0.35)
+    this.orientation = 4 // down
+    this.speed = this.this.game.playerSpeed
+    // TODO create dialogue logic.. maybe
+    // this.dialoguesMemory = {}
+    // this.maxLife = this.this.game.playerLife
+    // this.life = this.maxLife
+    this.inFight = false
+    this.defaultFrames = {
```
Comment (kskkido) [inline app/states/playerRefactor.js]: change to this.frames

File: app/states/playerRefactor.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+// client side class for Playable Characters
+export default class Player extends Prefab {
+  constructor(x, y, key) {
+    // call it's superclass constructor
+    super(x, y, key)
+    // Send context as first argument!!
+    this.anchor.set(0.25, 0.35)
+    this.orientation = 4 // down
+    this.speed = this.this.game.playerSpeed
```
Comment (kskkido) [inline app/states/playerRefactor.js]: remove extra this

File: app/states/playerRefactor.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    // this.addChild(this.weapon = this.this.game.add.sprite(0, 0, 'atlas3'))
+    // this.addChild(this.shadow = this.this.game.add.sprite(0, 5, 'atlas1', 'shadow'))
+    this.addChild(this.nameHolder = this.this.game.add.text(0, -30, '', {
+      font: '14px pixel',
+      fill: '#ffffff',
+      stroke: '#000000',
+      strokeThickness: 1
+    }))
+    this.events.onKilled.add(function(player) {
+      this.this.game.displayedPlayers.delete(player.id)
```
Comment (kskkido) [inline app/states/playerRefactor.js]: remove duplicate this

File: app/states/playerRefactor.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    if (this.isPlayer) this.nameHolder.addColor('#f4d442', 0)
+  }
+  // sets the player's name
+  setName(name) {
+    this.nameHolder.text = name
+    this.nameHolder.x = Math.floor(16 - (this.nameHolder.width/2))
+  }
+
+  // TODO refactor this to use cursor keys (since we won't be using clicks for movement)
+  prepareMovement(end, finalOrientation, action, delta, sendToServer) {
```
Comment (kskkido) [inline app/states/playerRefactor.js]: review this function

File: app/states/playerRefactor.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      Client.setWeapon(key)
+    }
+    return true
+  }
+
+  adjustWeapon() {
+    this.weapon.position.set(this.weapon.offsets.x, this.weapon.offsets.y)
+  }
+
+  equipArmor(key) {
```
Comment (kskkido) [inline app/states/playerRefactor.js]: take out frame and animation aspect of equipArmor

File: app/states/playerRefactor.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    const width = Game.computeLifeBarWidth()
+    const tweenWidth = this.game.add.tween(Game.health.getChildAt(0)) // tween for the "body" of the bar
+    const tweenEnd = this.game.add.tween(Game.health.getChildAt(1)) // tween for the curved tip
+    tweenWidth.to({ width: width }, 200, null, false, 200)
+    tweenEnd.to({ x: width }, 200, null, false, 200)
+    tweenWidth.start()
+    tweenEnd.start()
+  }
+
+  fight() {
```
Comment (kskkido) [inline app/states/playerRefactor.js]: remove fight and fight action functions - user players do not need these

File: app/states/playerRefactor.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    if (!this.target) return
+    this.inFight = true
+    this.fightTween = this.game.add.tween(this) // adds a tween to the game state
+    this.fightTween.to({}, Phaser.Timer.SECOND, null, false, 0, -1)
+    this.fightTween.onStart.add(function() { this.fightAction() }, this)
+    this.fightTween.onLoop.add(function() { this.fightAction() }, this)
+    this.fightTween.start()
+  }
+
+  fightAction() {
```
Comment (kskkido) [inline app/states/playerRefactor.js]: remove

File: app/states/playerRefactor.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+        this.tween = null
+      }
+      this.orientation = direction
+      this.attack()
+    }
+  }
+
+  die(animate) {
+    if (this.tween) this.stopMovement(false)
+    this.endFight()
```
Comment (kskkido) [inline app/states/playerRefactor.js]: remove this.endFight()

File: app/states/playerRefactor.js [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+      'down': [27, 30],
+      'idle_down': [31, 32],
+      'attack_left': [33, 37, 42],
+      'left': [38, 41],
+      'idle_left': [42, 43]
+    }
+    // TODO reassign these children based on our assets
+    // this.addChild(this.weapon = this.this.game.add.sprite(0, 0, 'atlas3'))
+    // this.addChild(this.shadow = this.this.game.add.sprite(0, 5, 'atlas1', 'shadow'))
+    this.addChild(this.nameHolder = this.this.game.add.text(0, -30, '', {
```
Comment (kskkido) [inline app/states/playerRefactor.js]: I think we can remove lines 36 to 41.  I will move it to entityPrefab

**Review by kskkido — CHANGES_REQUESTED** [review]

File: app/states/playerRefactor.js [hunk 10]
```diff
@@ -0,0 +1,155 @@
+import Prefab from './entityPrefab'
+
+// client side class for Playable Characters
+export default class Player extends Prefab {
+  constructor(x, y, key) {
+    // call it's superclass constructor
+    super(x, y, key)
```
Comment (kskkido) [inline app/states/playerRefactor.js]: Change constructor and super arguments:
constructor(game, name, position, spriteKey)
super(game, name, position, spriteKey)

**Review by kskkido — COMMENTED** [review]
one more change I missed on the first review

**Review by JonathanLiu25 — APPROVED** [review]

**Pull request merged**